src/test/java/com/coveo/feign/BaseServiceException.java

```diff
@@ -0,0 +1,30 @@
+package com.coveo.feign;
+
+public abstract class BaseServiceException extends Exception {
```

Comment:
why did you add this exception type ? It would have worked with current ServiceException no ? just curious